### DISCUSSION TITLE
Can't extend default TypeORM legacy adapter entities when using database config obj

### DISCUSSION THREAD

**Discussion by gtoshinakano** [nextauthjs/next-auth#4518] [Help] [answered]
### Environment

  System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
    Memory: 7.85 GB / 15.93 GB
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 100.0.4896.127
    Edge: Spartan (44.19041.1266.0), Chromium (101.0.1210.32)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    next: ^12.1.5 => 12.1.5
    next-auth: ^4.3.4 => 4.3.4
    react: ^17.0.2 => 17.0.2

### Reproduction URL

https://next-auth.js.org/adapters/typeorm

### Describe the issue

I'm trying to extend or override the default Entities classes on nextAuth v4 with a self-hosted MySQL DB and typeORM legacy adapter but it doesn't work. It uses de default module 

### How to reproduce

I tried on [...nextauth].js
```
import entities from '@Entities' //my custom entities 
const db = {
  type: process.env.DB_TYPE,
  host: process.env.DB_HOST,
  port: process.env.DB_PORT,
  username: process.env.DB_USER,
  password: process.env.DB_PASS,
  database: process.env.DB_NAME,
  synchronize: false
  logging: true, 
};
  ...

 adapter: TypeORMLegacyAdapter(db, {entities})
```

But it still uses the default Entities from @next-auth/typeorm-legacy-adapter

I tried the options below:
1- MySQL connection string 
```
 adapter: TypeORMLegacyAdapter('mysql://connection:string', {entities}) // doesn't connect even with the right credentials
```
2- Import and pass param as array
```
import {UserEntity, SessionEntity, ... } from '@Entities'
...
adapter: TypeORMLegacyAdapter(db, {entities: [UserEntity, SessionEntity, ...]}) 
```
3-Put entities on ```db``` object 
```
import entities from '@Entities' // tried all variations

const db = {
  type: process.env.DB_TYPE,
  host: process.env.DB_HOST,
  port: process.env.DB_PORT,
  username: process.env.DB_USER,
  password: process.env.DB_PASS,
  database: process.env.DB_NAME,
  synchronize: false
  logging: true, 
  entities
};
```

### Expected behavior

As per said in the documentation, I should be able to extend or override the custom TypeORM Entities.
Does anyone know how to do it properly?

**Comment by balazsorban44** [comment, marked as answer]
We have tests in place for this to be sure it works. You can have a look at https://github.com/nextauthjs/next-auth/tree/main/packages/adapter-typeorm-legacy/tests

to get an idea. Hope that helps.

Otherwise you can also, see the docs https://next-auth.js.org/adapters/typeorm#custom-models

**Comment by SemajDraw** [comment]
I am having the same issue. Passing any entities to the TypeORMLegacyAdapter, even the model it is using by default causes the following error to be thrown 
```
Entity metadata for AccountEntity#user was not found. Check if you specified a correct entity object and if it's connected in the connection options. TypeORMError: Entity metadata for AccountEntity#user was not found. Check if you specified a correct entity object and if it's connected in the connection options.
```

My NextAuth configuration
```
import { TypeORMLegacyAdapter } from '@next-auth/typeorm-legacy-adapter';
import NextAuth, { NextAuthOptions } from 'next-auth';
import GoogleProvider from 'next-auth/providers/google';
import { DataSourceOptions } from 'typeorm';
import * as entities from '../../../entities';

const postgresConfig: DataSourceOptions = {
  type: 'postgres',
  host: process.env.DB_HOST,
  port: parseInt(process.env.DB_PORT as string),
  username: process.env.DB_USERNAME,
  password: process.env.DB_PASSWORD,
  database: process.env.DB_NAME,
  synchronize: true,
};

export const authOptions: NextAuthOptions = {
  debug: true,
  adapter: TypeORMLegacyAdapter(postgresConfig, { entities }),
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_ID as string,
      clientSecret: process.env.GOOGLE_SECRET as string,
    }),
  ],
};

export default NextAuth(authOptions);
```

**Reply by andrisole92** [reply]
Having same problem.

Only happens in prod

```
https://next-auth.js.org/errors#adapter_error_getuserbyaccount No metadata for "AccountEntity" was found. {
  message: 'No metadata for "AccountEntity" was found.',
  stack: 'EntityMetadataNotFoundError: No metadata for "AccountEntity" was found.\n' +
    '    at DataSource.getMetadata (/Users/andrejsruz/Desktop/dev/relty/next-client/node_modules/typeorm/data-source/DataSource.js:299:19)\n' +
    '    at EntityManager.findOne (/Users/andrejsruz/Desktop/dev/relty/next-client/node_modules/typeorm/entity-manager/EntityManager.js:581:42)\n' +
    '    at getUserByAccount (/Users/andrejsruz/Desktop/dev/relty/next-client/.next/server/chunks/525.js:1:5388)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)',
  name: 'EntityMetadataNotFoundError'
}

```

**Reply by gkorland** [reply]
I'm getting the same exact error only in production.
@andrisole92 did you find a solution

**Reply by andrisole92** [reply]
@gkorland using TypeOrmLegacyAdapter.

**Reply by gkorland** [reply]
@andrisole92 do you know what is the source of the issue?

**Reply by andrisole92** [reply]
@gkorland i am not sure can be a few things.

**Reply by gkorland** [reply]
I tried the TypeOrmLegacyAdapter but it seems like with Next 13.5 I'm getting the same error again

**Reply by andrisole92** [reply]
This is my conf:

```
import { TypeORMLegacyAdapter } from '@next-auth/typeorm-legacy-adapter';

const adapter: Adapter = TypeORMLegacyAdapter({
	type: 'postgres',
	host: `${process.env.DATABASE_HOST}`,
	port: Number(process.env.DATABASE_PORT) ?? 5555,
	username: `${process.env.DATABASE_USERNAME}`,
	password: `${process.env.DATABASE_PASSWORD}`,
	database: `${process.env.DATABASE_NAME}`,
	synchronize: false,
	poolSize: 2,
	ssl: {
		rejectUnauthorized: false,
	},
});
```


You can alternatively try Prisma Adapter that works well with SQL, but they don't allow changing table names...
